core/common/src/ContextRealityModel.ts

```diff
@@ -22,10 +22,82 @@ export interface OrbitGtBlobProps {
   accountName: string;
 }
 
+/** Identify the Reality Data service provider
+ * @alpha
+ */
+export enum RealityDataProvider {
+  /**
+   * This is the legacy mode where the access to the 3d tiles is harcoded in ContextRealityModelProps.tilesetUrl property.
+   * It was use to support RealityMesh3DTiles, Terrain3DTiles, Cesium3DTiles
+   * You should use other mode when possible
+   * @see [[RealityDataSource.createRealityDataSourceKeyFromUrl]] that will try to detect provider from an URL
+   * */
+  TilesetUrl = "TilesetUrl",
+  /**
+   * Will provide access url from realityDataId and iTwinId on contextShare for 3dTile storage format or  OPC storage format
+   * This provider support all type of 3dTile storage fomat and OrbitPointCloud: RealityMesh3DTiles, Terrain3DTiles, Cesium3DTiles, OPC
+   * @see [[RealityDataFormat]].
+  */
+  ContextShare = "ContextShare",
+  /**
+   * Will provide Open Street Map Building (OSM) from Cesium Ion (in 3dTile format)
+  */
+  CesiumIonAsset = "CesiumIonAsset",
+}
+
+/** Identify the Reality Data storage format
+ * @alpha
+ */
+export enum RealityDataFormat {
+  /**
+   * 3dTile supported formats; RealityMesh3DTiles, Terrain3DTiles, Cesium3DTiles
+   * */
+  ThreeDTile = "ThreeDTile",
+  /**
+   * Orbit Point Cloud (OPC) storage format (RealityDataType.OPC)
+  */
+  OPC = "OPC",
+}
+
+/**
+ * Key used by ContextShare RealityDataProvider
+ * @alpha
+ */
+export interface RealityDataSourceKey {
+  /**
+   * The provider that supplies the access to reality data source for displaying the reality model
+   * @see [[RealityDataProvider]] for default supported value;
+  */
+  provider: string;
+  /**
+   * The format used by the provider to store the reality data
+   * @see [[RealityDataFormat]] for default supported value;
+  */
+  format: string;
+  /** The reality data id that identify a reality data for the provider */
+  id: string;
+  /** The context id that was used when reality data was attached - if none provided, current session iTwinId will be used */
+  iTwinId?: string;
+}
+
+/** JSON representation of the reality data reference attachment properties.
+ * @alpha
+ */
+export interface RealityDataSourceProps {
+  /** The source key that identify a reality data for the provider. */
+  sourceKey: RealityDataSourceKey;
+}
+
 /** JSON representation of a [[ContextRealityModel]].
  * @public
  */
 export interface ContextRealityModelProps {
+  /**
+   * The reality data source key identify the reality data provider and storage format.
+   * It takes precedence over tilesetUrl and orbitGtBlob when present and can be use to actually replace these properties.
+   * @alpha
+   */
+  rdSourceKey?: RealityDataSourceKey;
   /** The URL that supplies the 3d tiles for displaying the reality model. */
   tilesetUrl: string;
```

Comment:
It looks like this should now be optional if `rdSourceKey` is defined.

Reply:
Yes, that is why it is left required, to prevent from breaking API.